internal/api/verify.go

```diff
@@ -126,9 +126,13 @@ func (a *API) verifyGet(w http.ResponseWriter, r *http.Request) error {
 		if terr = a.setCookieTokens(config, token, false, w); terr != nil {
 			return internalServerError("Failed to set JWT cookie. %s", terr)
 		}
+		if params.Type == recoveryVerification {
+			if terr := user.LogoutAllSessionsExceptMine(tx); terr != nil {
+				return terr
+			}
+		}
```

Comment:
I think we should do this after issuing the tokens in [`verifyGet`](https://github.com/supabase/gotrue/blob/d0d079fb44c99fd1953e9bd8105c6350a4d9c330/internal/api/verify.go#L120) and [`verifyPost`](https://github.com/supabase/gotrue/blob/d0d079fb44c99fd1953e9bd8105c6350a4d9c330/internal/api/verify.go#L213). This is because `issueRefreshToken` will always create a new session on password recovery, so we can grab the session id created and use that to filter out all the other sessions. 

Also, maybe it's better to do this in the `updateUser` endpoint if the password is changed. WDYT?

Reply:
Sure, that makes sense - I'll move it there and amend the PR